infrastructure/merklemountainrange/src/merklemountainrange.rs

```diff
@@ -53,13 +58,24 @@ where
             data: HashMap::new(),
             hasher: PhantomData,
             current_peak_height: (0, 0),
+            _settings: MmrSettings::default(),
+            _states: Vec::new(),
         }
     }
 
+    #[cfg(debug_assertions)]
+    pub fn get_data_object(&self, hash: ObjectHash) -> Option<&MerkleNode<T>> {
+        self.data.get(&hash)
+    }
+
     /// This function returns a reference to the data stored in the mmr
     /// It will return none if the hash does not exist
     pub fn get_object(&self, hash: &ObjectHash) -> Option<&T> {
-        self.data.get(hash)
+        let object = self.data.get(hash);
+        if object.is_none() || !object.unwrap().pruned {
```

Comment:
```suggestion
        if object.is_none() || object.unwrap().pruned {
```

Isn't this the wrong way round?